src/main/java/org/jenkinsci/plugins/workflow/steps/IsMacStep.java

```diff
@@ -0,0 +1,65 @@
+package org.jenkinsci.plugins.workflow.steps;
+
+import hudson.Extension;
+import hudson.Launcher;
+import hudson.Platform;
+import hudson.remoting.VirtualChannel;
+import jenkins.security.MasterToSlaveCallable;
+
+import java.util.Collections;
+import java.util.Set;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+/**
+ * Checks whether we are running on Mac.
+ */
+public class IsMacStep extends Step {
+
+    @DataBoundConstructor public IsMacStep() {}
+
+    @Override public StepExecution start(StepContext context) throws Exception {
+        return new Execution(context);
+    }
+
+    public static class Execution extends SynchronousNonBlockingStepExecution<Boolean> {
```

Comment:
This differs from

https://github.com/jenkinsci/workflow-basic-steps-plugin/blob/ad0ec60819e306d9619768d2aecb72a144b61724/src/main/java/org/jenkinsci/plugins/workflow/steps/IsUnixStep.java#L44

Why?

Reply:
From the Javadoc of the `SynchronusStepExecution`
```
To call legacy Jenkins APIs which are potentially long-running and interruptible yet offer no asynchronous mode (for example because they block on a remoting call) use SynchronousNonBlockingStepExecution
```
I did it due to the remoting call.

Reply:
@jglick  
Would this mean that `isUnixStep` should also be a `SynchronousNonBlockingStepExecution`?

Reply:
What Remoting call? https://github.com/jenkinsci/jenkins/blob/8337315cced961853eb0d9f24034290066783662/core/src/main/java/hudson/Launcher.java#L1076-L1079